Compilation error

Hi! 

I'm using hls_m3u8 and as of recently I'm getting a compilation error:

```
error[E0759]: `input` has an anonymous lifetime `'_` but it needs to satisfy a `'static` lifetime requirement
   --> /Users/wouter.de.bie/.cargo/registry/src/github.com-1ecc6299db9ec823/hls_m3u8-0.4.0/src/media_playlist.rs:730:12
    |
729 |     fn from_str(input: &str) -> Result<Self, Self::Err> {
    |                        ---- this data with an anonymous lifetime `'_`...
730 |         Ok(parse_media_playlist(input, &mut Self::builder())?.into_owned())
    |            ^^^^^^^^^^^^^^^^^^^^^-----^^^^^^^-------------^^^
    |                                 |           |
    |                                 |           ...and is required to live as long as `'static` here
    |                                 ...is captured here...
```

My environment:
```
stable-x86_64-apple-darwin (default)
rustc 1.54.0 (a178d0322 2021-07-26)
```

Any idea?

--- a/src/media_playlist.rs
+++ b/src/media_playlist.rs
@@ -727,7 +727,7 @@ impl FromStr for MediaPlaylist<'static> {
     type Err = Error;
 
     fn from_str(input: &str) -> Result<Self, Self::Err> {
-        Ok(parse_media_playlist(input, &mut Self::builder())?.into_owned())
+        Ok(parse_media_playlist(input, &mut MediaPlaylist::builder())?.into_owned())
     }
 }
 

--- a/src/types/float.rs
+++ b/src/types/float.rs
@@ -12,7 +12,7 @@ use crate::Error;
 /// [`NaN`]: core::f32::NAN
 /// [`INFINITY`]: core::f32::INFINITY
 /// [`NEG_INFINITY`]: core::f32::NEG_INFINITY
-#[derive(AsRef, Deref, Default, Debug, Copy, Clone, Display, PartialOrd)]
+#[derive(AsRef, Deref, Default, Debug, Copy, Clone, Display)]
 pub struct Float(f32);
 
 impl Float {
@@ -128,6 +128,11 @@ impl PartialEq<f32> for Float {
 // with NaN so all of the above requirements are satisfied and therefore Eq can
 // be soundly implemented.
 impl Eq for Float {}
+
+impl PartialOrd for Float {
+    #[inline]
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> { Some(self.cmp(other)) }
+}
 
 impl Ord for Float {
     #[inline]

--- a/src/types/in_stream_id.rs
+++ b/src/types/in_stream_id.rs
@@ -19,8 +19,8 @@ use crate::types::ProtocolVersion;
 /// [`MediaPlaylist`]: crate::MediaPlaylist
 #[non_exhaustive]
 #[allow(missing_docs)]
-#[strum(serialize_all = "UPPERCASE")]
 #[derive(Ord, PartialOrd, Debug, Clone, Copy, PartialEq, Eq, Hash, Display, EnumString)]
+#[strum(serialize_all = "UPPERCASE")]
 pub enum InStreamId {
     Cc1,
     Cc2,

--- a/src/types/ufloat.rs
+++ b/src/types/ufloat.rs
@@ -13,7 +13,7 @@ use crate::Error;
 /// [`NaN`]: core::f32::NAN
 /// [`INFINITY`]: core::f32::INFINITY
 /// [`NEG_INFINITY`]: core::f32::NEG_INFINITY
-#[derive(AsRef, Deref, Default, Debug, Copy, Clone, PartialOrd, Display)]
+#[derive(AsRef, Deref, Default, Debug, Copy, Clone, Display)]
 pub struct UFloat(f32);
 
 impl UFloat {
@@ -140,6 +140,11 @@ impl PartialEq<f32> for UFloat {
 // with NaN so all of the above requirements are satisfied and therefore Eq can
 // be soundly implemented.
 impl Eq for UFloat {}
+
+impl PartialOrd for UFloat {
+    #[inline]
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> { Some(self.cmp(other)) }
+}
 
 impl Ord for UFloat {
     #[inline]

